crates/bevy_render/src/view/visibility/mod.rs

```diff
@@ -34,19 +34,40 @@ pub struct Visibility {
 
 impl Default for Visibility {
     fn default() -> Self {
+        Self::visible()
+    }
+}
+
+impl Visibility {
+    /// Creates a new [`Visibility`], set as visible
+    pub const fn visible() -> Self {
         Self { is_visible: true }
     }
 }
 
 /// Algorithmically-computed indication of whether an entity is visible and should be extracted for rendering
-#[derive(Component, Clone, Reflect, Debug, Eq, PartialEq, Default)]
+#[derive(Component, Clone, Reflect, Debug, Eq, PartialEq)]
 #[reflect(Component)]
 pub struct ComputedVisibility {
     is_visible_in_hierarchy: bool,
     is_visible_in_view: bool,
 }
 
+impl Default for ComputedVisibility {
+    fn default() -> Self {
+        Self::visible()
```

Comment:
Are you sure there is no code that relies on ComputedVisibility::default() being false? I think they have to be reset at the beginning of each check_visibility or so, so maybe it gets taken care of. But like GlobalTransform, it's not really valid until after the visibility checks and propagation have been done.

Reply:
Reset to `false` that is.

Reply:
Yeah the default values here don't matter "functionality-wise" other than users reading the initial state during the first frame prior to the visibility checks running in PostUpdate. There _might_ be valid use cases that want to read "last frame" visibility logic. And then theres also the case of people accidentally reading this too early (probably more common). In both cases, I think it is much safer to default to "invisible". Imagine spawning a new scene with various entities invisible and visible. Is it better for the first frame logic to consider _everything_ in the scene (visibility defaulting to true) or _nothing_ in the scene (visibility defaulting to false). Imo it should be "nothing". If this early-read-logic fed in to rendering logic (generally it wouldn't, but I think it helps make the point), we would absolutely prefer skipping a frame before rendering (invisible by default) rather than than rendering stuff we shouldn't (visible by default).

Reply:
Also logically it makes sense. Last frame, was a newly spawned entity visible or invisible? Invisible seems correct to me.

Reply:
ok!